transformer_lens/components.py

```diff
@@ -350,38 +355,95 @@ def forward(
         past_kv_cache_entry is an optional entry of past keys and values for this layer, only relevant if generating text. Defaults to None
 
         """
+        # If user requests it, make a hookpoint with separate inputs for each head.
+        # This allows the user to change the inputs to different heads independently.
+        if self.cfg.use_attn_input:
+            input_to_q = self.hook_input_q(
+                resid_pre.unsqueeze(2).expand(-1, -1, self.cfg.n_heads, -1).clone()
```

Comment:
Please rewrite to einops

`einops.repeat(resid_pre, "batch pos d_model -> batch pos head_index d_model", head_index=self.cfg.n_heads)` should work and IMO is much more readable